plugins/traction_innkeeper/traction_innkeeper/v1_0/innkeeper/routes.py

```diff
@@ -522,6 +532,60 @@ async def tenant_create_token(request: web.BaseRequest):
     return web.json_response({"token": token})
 
 
+@docs(
+    tags=["multitenancy"],
+    summary="Get auth token for a subwallet (innkeeper plugin override)",
+)
+@request_schema(CreateWalletTokenRequestSchema)
+@response_schema(CreateWalletTokenResponseSchema(), 200, description="")
+@error_handler
+async def tenant_wallet_create_token(request: web.BaseRequest):
```

Comment:
So for discussion and any other people's thoughts on what we are doing here (correct me if I'm wrong on any of this)

- We want to stop token creation for "suspended" soft-deleted tenants (IE tenants in the deleted state)
- Token creation is overridden in the other plugin Multitenant Provider
- See that here https://github.com/hyperledger/aries-acapy-plugins/blob/main/multitenant_provider/multitenant_provider/v1_0/routes.py#L76
- We can't put the tenant state check there because in that plugin there isn't any "Tenant" concept, that's Innkeeper plugin and obviously the multitenant provider could be used independently (why use that plugin, see details here https://github.com/hyperledger/aries-acapy-plugins/tree/main/multitenant_provider)

I guess one concern of this here is we are duplicating some code essentially, if some enhancement or fix happens in the other plugin create_token we wouldn't get it here?

There's some other code in the plugin at this line:

https://github.com/hyperledger/aries-acapy-plugins/blob/main/multitenant_provider/multitenant_provider/v1_0/routes.py#L116

where `if (not wallet_record.requires_external_key) and wallet_key:` begins.
That's not replicated here so are we bypassing that logic then? The `always_check_provided_wallet_key` check there... I'm not sure if that's something necessary, need to refresh my memory and test out what it's doing.

Regardless, duplicating this function here in Innkeeper might be necessary for the pattern of preventing tokens for deleted tenants, but if anyone else has any thoughts/opinions can discuss here

Reply:
Off the top of my head, I think the duplication might be inevitable since we overriding multiple times the same function and there is - in theory - no guarantee that the plugins would be installed and executed in the same order. Which brings me to the follow-up question: would it cause issues to have the plugins installed in reverse order than what it was tested?

i.e.: currently we install multitenant provider and then innkeeper, with multitenant overriding built-in endpoints and innkeeper overriding the result of that. What happens if we reverse the order of installation? If we must install plugins in a specific order we may want to add a note in the docs about that.

Reply:
We need to explicitly load multitenant before innkeeper for these overrides to function

Reply:
I have now imported the original function used in the multitenant plugin inside of the innkeeper plugin to reduce the duplication of code
https://github.com/bcgov/traction/blob/block-api-for-deleted-tenant/plugins/traction_innkeeper/traction_innkeeper/v1_0/innkeeper/routes.py#L557